mevislab.github.io/content/tutorials/thirdparty/thirdpartyexample1.md

```diff
@@ -0,0 +1,210 @@
+---
+title: "Example 1: OpenCV Webcam access"
+date: 2022-06-15T08:56:33+02:00
+status: "open"
+draft: false
+tags: ["Advanced", "Tutorial", "OpenCV", "Python", "Webcam", "Macro", "Macro Modules", "Global Macro"]
+menu: 
+  main:
+    identifier: "thirdpartyexample1"
+    title: "Access your webcam and use the live video in MeVisLab via OpenCV."
+    weight: 805
+    parent: "thirdparty"
+---
+# Example 1: OpenCV Webcam access
+
+## Introduction
+In this example, we are using the `PythonImage` module and access your webcam to show the video in a `View2D`.
+
+## Steps to do
+### Creating the network to be used for testing
+Add the modules to your workspace and connect them as seen below. 
+
+![Example Network ](/images/tutorials/thirdparty/network_example1.png "Example Network ")
+
+The viewer is empty because the image needs to be set via Python scripting.
+
+{{<alert class="info" caption="Info">}}
+More information about the `PythonImage` module can be found {{< docuLinks "/Standard/Documentation/Publish/ModuleReference/PythonImage.html" "here" >}}
+{{</alert>}}
+
+### Create a Macro Module
+Now you need to create a Macro Module from your network. You can either group your modules, create a local Macro and convert it to a global Macro Module, or you use the Project Wizard and load your *.mlab file.
+
+{{<alert class="info" caption="Info">}}
+A tutorial how to create your own Macro Modules can be found in [Example 2.2: Global Macro Modules](/tutorials/basicmechanisms/macromodules/globalmacromodules "Example 2.2: Global Macro Modules"). Make sure to add a Python file to your Macro Module.
+{{</alert>}}
+
+### Add the View2D to your UI
+Next, we need to add the `View2D` to a Window of your Macro Module. Right click on your module {{< mousebutton "right" >}}, open the context menu and select {{< menuitem "Related Files" "<YOUR_MODULE_NAME>.script" >}}. The text-editor {{< docuLinks "/Resources/Documentation/Publish/SDK/MeVisLabManual/ch26.html" "MATE">}} opens. You can see the file *.script file of your module.
+
+Add the following to your file:
+{{< highlight filename="<YOUR_MODULE_NAME>.script" >}}
+```Stan
+Interface  {
+  Inputs {}
+  Outputs {}
+  Parameters {}
+}
+
+Commands {
+  source = $(LOCAL)/<YOUR_MODULE_NAME>.py
+}
+
+Window {
+  h= 500
+  w= 500
+  initCommand = setupInterface
+  destroyedCommand = releaseCamera
+  
+  Vertical {
+    Horizontal {
+      Button {
+        title = Start
+        command = startCapture
+      }
+      Button {
+        title = Pause
+        command = stopCapture
+      }
+    }
+    Horizontal {
+      expandX = True
+      expandY = True
+      
+      Viewer View2D.self {
+        type = SoRenderArea
+      }
+    }
+  }
+}
+```
+{{</highlight>}}
+
+Now open the Python file of your module and define the commands to be called from the *.script file:
+{{< highlight filename="<YOUR_MODULE_NAME>.py" >}}
+```Python
+# from mevis import *
```

Comment:
These are only the commands, in case a user wants to open the Window, there will be Python errors because of missing functions. I thought it is a good idea to add the functions initially.